### ISSUE TITLE
api.MediaDevices.enumerateDevices - doesnt return audio output device in ios safari

### ISSUE THREAD

**Issue by Ezhil-Git** [mdn/browser-compat-data#23148]
### What type of issue is this?

Missing compatibility data

### What information was incorrect, unhelpful, or incomplete?

Safari is not fully supporting the enumerateDevices, even after getUserMedia is allowed, audiooutput device is 0 

### What browsers does this problem apply to, if applicable?

Safari

### What did you expect to see?

With headphones connected, i must be able to see external headphone and built in speaker listed

### Did you test this? If so, how?

https://jsfiddle.net/px56shgz/

### Can you link to any release notes, bugs, pull requests, or MDN pages related to this?

https://stackoverflow.com/questions/62827607/audio-output-device-array-is-of-length-0-on-safari

### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/enumerateDevices

### MDN metadata

<details>
<summary>MDN page report details</summary>

* Query: `api.MediaDevices.enumerateDevices`
* Report started: 2024-05-22T08:43:44.872Z

</details>

**Comment by juancholt** [comment]
This is also happening on desktop safari

**Comment by ornakash** [comment]
Same here

**Comment by nguyenminhnh4t** [comment]
I got the same problem

**Comment by maitogami123** [comment]
I also have the same problem

**Comment by conradfr** [comment]
Still a thing.

**Comment by brokeboiflex** [comment]
From my tests, the reason it happens is becouse the audio input and audio output are coupled together - you can reproduce it quite easily by connecting any bluetooth headphones and toggling between iphoneMic and headphones as an audio source.

TLDR: **use audio input devices as audio output devices** and modify the tracks as needed. I know it's bad dev experience, but there's nothing that can be done about apple's perpetual autism.


Edit: 
Bugzilla confirms the results of my tests https://bugs.webkit.org/show_bug.cgi?id=179415
